docs/user-guide/administration/design-table.md

```diff
@@ -19,220 +19,257 @@ Before proceeding, please review the GreptimeDB [Data Model Documentation](/user
+  For example, `trace_id`, `span_id`, `user_id`, `uri`, `ip`, `uuid`, `request_id`, table auto increment id are usually high cardinality.
 
 ## Column Types and Selection
 
-In GreptimeDB, columns are categorized into three types: Tag, Field, and Time Index.
-The timestamp usually represents the time of data sampling or the occurrence time of logs/events and is used as the Time Index column.
-GreptimeDB optimizes data organization based on the Time Index to enhance query performance.
+In GreptimeDB, columns are categorized into three semantic types: `Tag`, `Field`, and `Timestamp`.
+The timestamp usually represents the time of data sampling or the occurrence time of logs/events.
+GreptimeDB uses the `TIME INDEX` constraint to identify the `Timestamp` column.
+So the `Timestamp` column is also called the `TIME INDEX` column.
+If you have multiple columns with timestamp data type, you can only define one as `TIME INDEX` and others as `Field` columns.
 
-### Tag Columns
+In GreptimeDB, tag columns are optional. The main purposes of Tag columns include:
 
-Tag columns, also known as label columns,
-generally carry metadata of the measured data or logs/events.
-For example, when collecting nationwide weather temperature data,
-the city (e.g., `city="New York"`) is a typical tag column.
-In monitoring, system metrics like CPU and memory usually involve a `host` tag to represent the hostname.
+1. Defining the ordering of data in storage.
+   GreptimeDB reuses the `PRIMARY KEY` constraint to define tag columns and the ordering of tags.
+   Unlike traditional databases, in GreptimeDB, primary key and time series are identical.
+   Tables in GreptimeDB sort rows by (primary key, timestamp).
+   This improves the locality of data with the same tags.
+   If there are no tag columns, GreptimeDB sorts rows by timestamp.
+2. Identifying a unique time-series.
+   When the table is not append-only, GreptimeDB can deduplicate rows by timestamp under the same time-series (primary key).
+3. Smoothing migration from other TSDBs that use tags or labels.
 
-The main purposes of Tag columns in GreptimeDB include:
 
-1. Storing low-cardinality metadata.
-2. Filtering data, such as using the `city` column to view the average temperature in New York City over the past week. This is similar to the `WHERE` clause in SQL.
-3. Grouping and aggregating data. For instance, if the temperature data includes a `state` label in addition to `city`, you can group the data by `state` and calculate the average temperature for each `state` over the past week. This is similar to the `GROUP BY` clause in SQL.
+### Choosing tag columns
 
-Recommendations for Tag columns:
+Usually you don't need tag columns since ordering data by timestamp is sufficient for most use cases.
+If you need deduplication, or your queries can benefit from the ordering, you can define tag columns.
+Setting tags may also reduce disk space usage as it improves the locality of data.
```

Comment:
I feel this is due to the limitation caused cardinality effect on memtable. This should not be, at least the final, definition of tags. 

The selection of tags determines:

1. how we can dedup data
2. how we can sort data to narrow down the scan range of user's most typical queries
3. how we can do effective partition, for small query there should be less partitions envolved.
4. sorted data brings better compression as mentioned

Reply:
Removing the limitation may be the final goal, but this document is a guide for the current version.

However, a high cardinality primary key may still cause issues in some time-series workloads, like fetching last rows, processing in per-series manner. This may be related to the schema design.

Reply:
Not only memtable, our batch struct is also per series.
https://github.com/GreptimeTeam/greptimedb/blob/c77ce958a3b936e1065abe51fa10832b0ea2ac99/src/mito2/src/read.rs#L63-L74